.github/workflows/ci-docs.yml

```diff
@@ -14,14 +14,17 @@ on:
       - latest
   workflow_dispatch:
 
+permissions:
+  contents: read
+
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
       # Checkout the npm/cli repo
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5 # v2
```

Comment:
I’m not sure this has much value; this is an explicitly versioned tag by github themselves. If they made any breaking changes, it would break most of the projects hosted on github.

Reply:
GitHub's own repository pin's their checkout actions by SHA and doesn't use the version tag https://github.com/github/docs/blob/ea7f218c91ecbae9a700a8702b51a7d2736e0d2c/.github/workflows/docs-review-collect.yml#L23

Reply:
That just means they have an unnecessarily paranoid security team, not that it’s a reasonable thing to do. If they thought sha pinning was a good idea, they wouldn’t provide tags.